back/src/api/authentication/register/validator.ts

```diff
@@ -1,48 +1,30 @@
-import { StringSchema } from 'joi';
+import { z } from 'zod';
 import { buildValidationMiddleware } from '../../../core/middlewares';
-import { validator } from '../../../core/validator';
 import { config } from '../../../loader/config';
 import { emailDouaneRegex, passwordRegex } from '../common/const/regex';
 
-export interface IRegisterRequest {
-  body: {
-    email: string;
-    password: string;
-  };
-}
+export const registerValidator = z.object({
+  body: z.object({
+    email: z.string().refine(
```

Comment:
Il manque la gestion de l'erreur email requis
```suggestion
    email: z.string({ required_error: "L'email est requis" }).refine(
```